### PR TITLE
:sparkles: Add agent install namespace for addon deployment config

### PR DESCRIPTION
**Pull request by zhujian7**
## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by zhujian7** [comment]
@qiujian16

File: addon/v1alpha1/types_addondeploymentconfig.go [hunk 1]
```diff
@@ -50,6 +50,10 @@ type AddOnDeploymentConfigSpec struct {
 	// Empty means no proxy settings is available.
 	// +optional
 	ProxyConfig ProxyConfig `json:"proxyConfig,omitempty"`
+
+	// AgentInstallNamespace is the namespace where the add-on agent should be installed on the managed cluster.
+	// +optional
+	AgentInstallNamespace string `json:"agentInstallNamespace,omitempty"`
```
Comment (qiujian16) [inline addon/v1alpha1/types_addondeploymentconfig.go:59]: should we add a default value e.g. open-cluster-management-agent-addon?

Comment (zhujian7) [reply addon/v1alpha1/types_addondeploymentconfig.go:59]: ok, fixed

**Pull request merged**